### PR TITLE
cleanup: remove hadolint, cst and jx-release-version from the image

### LINKED ISSUE #62: Cleanup cst and hadolint
Let's cleanup cst and haodlint on subsequent PRs

And also jx-release-version (no usage outside release and docker pipeline library - [github.com/search?p=2&q=org%3Ajenkins-infra+jx-release-version&type=Code](https://github.com/search?p=2&q=org%3Ajenkins-infra+jx-release-version&type=Code))

_Originally posted by @dduportal in https://github.com/jenkins-infra/docker-builder/pull/59#pullrequestreview-1002683922_

### PR DESCRIPTION
**Pull request by lemeurherve**
Fixes #62

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -75,20 +56,14 @@ RUN mkdir -p /tmp/netlify && \
   && rm -rf /tmp/netlify /tmp/netlify.tar.gz \
   && netlify-deploy --help
 
-COPY --from=jx-release-version /usr/bin/jx-release-version /usr/bin/jx-release-version
-
 ## Repeating the ARGs from top level to allow them on this scope
```
Comment (lemeurherve) [inline Dockerfile:59]: Not quite sure we have to keep this repetition now there isn't any COPY anymore.

Comment (dduportal) [reply Dockerfile:59]: yup, you're right

File: Dockerfile [hunk 2]
```diff
@@ -75,20 +56,14 @@ RUN mkdir -p /tmp/netlify && \
...
 # Ref - https://docs.docker.com/engine/reference/builder/#scope
```
Comment (lemeurherve) [inline Dockerfile:60]:
```suggestion
```

File: Dockerfile [hunk 3]
```diff
@@ -75,20 +56,14 @@ RUN mkdir -p /tmp/netlify && \
...
-ARG JX_RELEASE_VERSION=2.5.2
 ARG JENKINS_AGENT_VERSION=4.13-2
 ARG ASDF_VERSION=0.8.1
```
Comment (lemeurherve) [inline Dockerfile:63]:
```suggestion
```

**Review by dduportal — APPROVED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -1,9 +1,6 @@
...
```
Comment (dduportal) [inline Dockerfile:2]: These arguments have to be moved after the `FROM` instruction

Comment (dduportal) [reply Dockerfile:2]: ref. https://docs.docker.com/engine/reference/builder/#scope

**Pull request merged**